docs/products/clickhouse/concepts/indexing.rst

```diff
@@ -1,43 +1,58 @@
 
 ClickHouse primary index
 ------------------------
 
-To find a specific block, ClickHouse uses **a primary index**. However, unlike in other systems the primary index in ClickHouse has a couple of distinctive characteristics.
+ClickHouse reads data in blocks. To find a specific block, ClickHouse uses its primary index, which has a few distinctive features that makes it different from primary indexes of the other systems.
+
+Instead of indexing every row, ClickHouse indexes very 10000th row (or, to be precise, every 8192nd row when using default settings). Such type of index is descriptively called *sparse index* and the batch of rows is sometimes called *granule*.
 
-First of all, not every row is indexed. By contrast, ClickHouse indexes every 10000th row (or, to be precise, the **8192nd**, when using default settings). Such type of index is descriptively called **a sparse index** and a batch of rows is sometimes called **a granule**.
+To make sparse indexing possible, ClickHouse sorts items according to the primary key. The items are intentionally sorted physically on the disk to speed up reading and prevent jumping across the disk when processing data. Therefore, by selecting a primary key, you determine how items are sorted physically on the disk. Such an approach helps ClickHouse effectively work on regular hard drives and depend less on SSD in comparison to other DBMSs.
 
-To make **sparse indexing** possible, ClickHouse sorts items according to the primary key. In fact, items are intentionally **sorted physically** on disk. This is done to speed up reading and to prevent jumping across the disk when processing data. Therefore, by selecting a primary key we determine how items are sorted physically on disk. Such an approach helps ClickHouse effectively work on regular hard drives and depend less on SSD in comparison to other DBMS.
+.. note::
+    
+    Even though ClickHouse sorts data by primary key, it is possible `to choose a primary key that is different from the sorting key <https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree/#choosing-a-primary-key-that-differs-from-the-sorting-key>`_.
 
-.. note:: Even though ClickHouse sorts data by primary key, it is possible `to choose a primary key that is different than the sorting key <https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree/#choosing-a-primary-key-that-differs-from-the-sorting-key>`_.
+Using sparse indexing has significant consequences for capabilities and limitations of ClickHouse. A primary key as used in ClickHouse do not ensure uniqueness for a single searched item since only every ten thousandth item is indexed. Thus, you need to iterate over thousands of items to find a specific row, which makes this approach inadequate when working with individual rows and suitable for processing millions or trillions of items.
```

Comment:
```suggestion
Using sparse indexing has significant consequences for the capabilities and limitations of ClickHouse. A primary key, as used in ClickHouse, does not ensure uniqueness for a single searched item since only every ten thousandth item is indexed. Thus, you need to iterate over thousands of items to find a specific row, which makes this approach inadequate when working with individual rows and suitable for processing millions or trillions of items.
```

Reply:
For "the", I'm skipping it for purpose since I dont mean any specific set of  CH capabilities and limitations that the user is familiar with, just referring to them in general.